lib/views/auth/join_verify_OTP_view.dart

```diff
@@ -0,0 +1,227 @@
+                  validator: ValidationUtils().validateOTP,
+                  onChanged: (String _) => setState(() => {}),
+                ),
+                Padding(
+                  padding: const EdgeInsets.symmetric(vertical: 8),
+                  child: _ResendCodeWidget(
+                    email: widget.email,
+                  ),
+                ),
+                Spacer(),
+                Row(
+                  mainAxisAlignment: MainAxisAlignment.end,
+                  children: [
+                    LoadingButtonWidget(
+                      onPressed: () => context.pop(),
+                      btnText: StringConstants.goBack,
+                    ),
+                    width8,
+                    LoadingButtonWidget(
+                      onPressed: _otpTextEditingController.text != ''
+                          ? _handleVerify
+                          : null,
+                      btnText: StringConstants.verify,
+                      bgColor: ColorConstants.walterWhite,
+                      textColor: ColorConstants.greyIsTheNewGrey,
+                      isLoading: isLoading,
+                    ),
+                  ],
+                ),
+                height8,
+              ],
+            ),
+          ),
+        ),
+      ),
+    );
+  }
+}
+
+class _ResendCodeWidget extends ConsumerWidget {
+  const _ResendCodeWidget({required this.email});
+  final String email;
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    var textTheme = Theme.of(context).textTheme;
+    var auth = ref.watch(authProvider);
+
+    void _handleResendOTP() async {
+      await auth.sendOTP(email);
+      auth.setCounter();
+      var status = auth.sendOTPRes.status;
+      if (status == Status.COMPLETED) {
+        showSnackBar(context, auth.sendOTPRes.body.toString());
+      } else if (status == Status.ERROR) {
+        showSnackBar(context, auth.sendOTPRes.message.toString());
+        auth.setCounter();
+      }
+    }
+
+    if (auth.sendOTPRes.status == Status.LOADING) {
+      return SizedBox(
+        height: 16,
+        width: 16,
+        child: CircularProgressIndicator(
+          strokeWidth: 2,
+          color: ColorConstants.walterWhite,
+        ),
+      );
+    }
+
+    return GestureDetector(
+      onTap: () => {},
+      child: Row(
+        children: [
+          GestureDetector(
+            onTap: _handleResendOTP,
+            child: Text(
+              auth.counter
+                  ? StringConstants.resendCodeIn
+                  : StringConstants.resendCode,
+              style: textTheme.bodyMedium?.copyWith(
+                color: ColorConstants.walterWhite,
+                fontFamily: DmSans,
+                height: 1.5,
+                fontSize: 14,
+                decoration: auth.counter
+                    ? TextDecoration.none
+                    : TextDecoration.underline,
+              ),
+            ),
+          ),
+          if (auth.counter)
```

Comment:
Hey @michaelcspeed, we can't use the brackets with if-statements and for for-loops. I extracted the widget as a method now it's more readable. I hope it will help.

` if (auth.counter) _counterAnimation(auth, textTheme),`